src/main/java/com/pirimid/cryptotrade/helper/exchange/coinbase/ExcCoinbase.java

```diff
@@ -0,0 +1,140 @@
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.List;
+
+@Component
+public class ExcCoinbase implements ExcParent {
+    @Value("${api.exchange.coinbase.baseurl}")
+    private String baseUrl;
+
+    private ResponseEntity<String> apiCallerOpen(String uri) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("Content-Type", "application/json")
+                .method("GET", HttpRequest.BodyPublishers.noBody())
+                .build();
+        HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity<String>(response.body(), headers, response.statusCode());
+    }
+
+    private ResponseEntity<String> apiCallerRestricted(String uri, String method, String apiKey, String passphrase, String signature, String timestamp, String body) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("Content-Type", "application/json")
+                .setHeader("CB-ACCESS-KEY", apiKey)
+                .setHeader("CB-ACCESS-PASSPHRASE", passphrase)
+                .setHeader("CB-ACCESS-SIGN", signature)
+                .setHeader("CB-ACCESS-TIMESTAMP", timestamp)
+                .method(method, HttpRequest.BodyPublishers.ofString(body))
+                .build();
+        HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity<String>(response.body(), headers, response.statusCode());
+    }
+
+    @Override
+    public List<SymbolResDTO> getPairs() {
+        SymbolResCoinbaseDTO symbolResCoinbaseDTO;
+        Gson gson = new Gson();
+        List<SymbolResCoinbaseDTO> symbolResCoinbaseDTOS;
+        List<SymbolResDTO> symbolResDTOS;
+        try {
+            ResponseEntity<String> response = apiCallerOpen(baseUrl + "/products");
+            Type SymbolListType = new TypeToken<List<SymbolResCoinbaseDTO>>() {
+            }.getType();
+            symbolResCoinbaseDTOS = gson.fromJson(response.getBody(), SymbolListType);
+            symbolResDTOS = CoinbaseUtil.getPairsResDTO(symbolResCoinbaseDTOS);
+            return symbolResDTOS;
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public ResponseEntity<String> accountInfo(String apiKey, String secretKey, String passphrase) {
+
+        String timestamp = Instant.now().getEpochSecond() + "";
```

Comment:
why empty double quote?

Reply:
just converted that to string. We can also use String.valueOf().